### ISSUE TITLE
If sound hasn't loaded and is played, the program crashes

### ISSUE THREAD

**Issue by MikaalSky** [Khan/live-editor#223]
Example: https://www.khanacademy.org/computer-programming/target-shooting-sound/5043290630782976

Click immediately after loading and keep clicking. The UI will half disappear.

**Comment by pamelafox** [comment]
Can you open your Chrome Dev Tools when it happens and tell me what the error is? I haven't managed to click it fast enough yet.

**Comment by MikaalSky** [comment]
InvalidStateError: An attempt was made to use an object that is not, or is no longer, usable scratchpads-exec-pjs-package-ddf410.js:7287

**Comment by kevinb7** [comment]
We probably want to do something similar to how images are loaded where there's callbacks for ```onload``` and ```onerror```.  HTMLAudioElements don't have respond to ```onload```.  Here's a link to an MDN page with a list of the events: https://developer.mozilla.org/en-US/docs/Web/Guide/Events/Media_events.  We probably want ```oncanplaythrough```.

**Comment by kevinb7** [comment]
I haven't been able to reproduce this error.  I tried to recreate it in isolation with a small example that sets audioElement.src and then calls audioElement.play() synchronously.  It appears that Chrome (	41.0.2272.35 on Mac OS X) queues ```play()``` calls and starts playing after the file has loaded.  I even tried simulating poor network connectivity to see if that would cause the InvalidStateError, no luck.

I did some Googling about InvalidStateError.  This issue usually occurs when the element isn't ready  for whatever reason.  People suggest using the following pattern to deal with the issue:
```
if (domObject.readyState > 0){
    // do something with domObject
  }
else{
    domObject.onload = function(e){  // or "onopen" or "oncanplaythrough"... whatever makes sense
        // do something with domObject
    }
}
```
Since we can attach the ```oncanplaythrough``` handler before we set the ```src``` the above pattern can be simplified.

TL;DR: I can't repro the bug, but I'm going to add an ```oncanplaythrough``` handler and do something similar to how images are loaded.

**Comment by kevinb7** [comment]
Fixed by https://github.com/Khan/live-editor/pull/243.
